### PR TITLE
Removing typying dependency

### PR DESCRIPTION
**Pull request by JunCEEE**
- Add SimEx-Lite and McStasScript tests to the CI
- Fix GitHub CI
- Remove typying dependency

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +25,26 @@ jobs:
...
+          cd SimEx-Lite
+          pip install -e .
+          git clone https://github.com/PaNOSC-ViNYL/SimEx-Lite-testFiles testFiles
+
+          git clone -b simex https://github.com/JunCEEE/pysingfel.git
+          cd pysingfel
+          pip install ./
+          cd ..
+          
+          pytest tests
```
Comment (mads-bertelsen) [inline .github/workflows/ci.yml:47]: Nice that projects that depend on libpyvinyl are tested when libpyvinyl is changed.

File: tests/integration/plusminus/requirements_dev.txt [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-sphinx_rtd_theme==0.5.1
+bump2version
+wheel
+watchdog
+flake8
+tox
+coverage
+Sphinx
+twine
+sphinx_rtd_theme
```
Comment (mads-bertelsen) [inline tests/integration/plusminus/requirements_dev.txt:10]: Great to drop the pinning!

**Review by mads-bertelsen — COMMENTED** [review]
We still have two dependency limits:
pint <= 0.19.2
dill <= 0.3.5.1

Are these still necessary?

Otherwise everything looks good!

**Comment by JunCEEE** [comment]
> We still have two dependency limits: pint <= 0.19.2 dill <= 0.3.5.1
> 
> Are these still necessary?
> 
> Otherwise everything looks good!

They are actually necessary. `pint <= 0.19.2` for pint.quantity. `dill <= 0.3.5.1` for dump.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -24,7 +25,26 @@ jobs:
           pytest .
       - name: Test McStasScript
         run: |
-          python3 -m pip install McStasScript
+          git clone https://github.com/PaNOSC-ViNYL/McStasScript.git
```
Comment (CFGrote) [inline .github/workflows/ci.yml:28]: could use `pip install git+https://...`

File: libpyvinyl/BaseCalculator.py [hunk 4]
```diff
@@ -49,7 +49,7 @@
 
 class BaseCalculator(AbstractBaseClass):
     """
-    :class:  Base class of all calculators.
```
Comment (CFGrote) [inline libpyvinyl/BaseCalculator.py:52]: why?

File: libpyvinyl/BaseCalculator.py [hunk 5]
```diff
@@ -49,7 +49,7 @@
...
+    Base class of all calculators.
 
     This class is provides the libpyvinyl API. It defines all methods
```
Comment (CFGrote) [inline libpyvinyl/BaseCalculator.py:54]: This class ~is~provides ...

File: libpyvinyl/BaseFormat.py [hunk 6]
```diff
@@ -4,7 +4,10 @@
 
 
 class BaseFormat(AbstractBaseClass):
-    """The abstract format class. It's the interface of a certain data format."""
```
Comment (CFGrote) [inline libpyvinyl/BaseFormat.py:7]: The abstract format class which serves as the common interface for derived format classes.

File: tests/integration/plusminus/requirements_dev.txt [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
-bump2version==0.5.11
-wheel==0.33.6
-watchdog==0.9.0
-flake8==3.7.8
-tox==3.14.0
-coverage==4.5.4
-Sphinx==3.5.2
-twine==1.14.0
-sphinx_rtd_theme==0.5.1
+bump2version
```
Comment (CFGrote) [inline tests/integration/plusminus/requirements_dev.txt:2]: why did you remove the version numbers? does this work on multiple OS/platforms?

File: libpyvinyl/BaseCalculator.py (hunk 5 above)
Comment (JunCEEE) [reply libpyvinyl/BaseCalculator.py]: Done.

File: libpyvinyl/BaseFormat.py (hunk 6 above)
Comment (JunCEEE) [reply libpyvinyl/BaseFormat.py:7]: Done.

File: tests/integration/plusminus/requirements_dev.txt (hunk 7 above)
Comment (JunCEEE) [reply tests/integration/plusminus/requirements_dev.txt]: Added the versions back.

**Review by CFGrote — APPROVED** [review]
